File: language-testing/fibonacci/fibonacci.pl
#! /usr/bin/env perl
# Prints out numbers from the fibonacci sequence
# Example usage: ./fibonacci.pl
# Example usage: ./fibonacci.pl 42

use strict;
use warnings;

sub fib {
  my $oldval = 1;
  my $newval = 0;
  my $tmp = 0;
  for (1..$_[0]) {
    $tmp = $oldval;
    $oldval = $newval;
    $newval = $tmp + $oldval;
  }
  return $newval;
}

# Fib the arugment, if possible:
# Otherwise, run 20 fibs
unless (caller) {
  if ($#ARGV == 0) {
    printf "%d\n", &fib($ARGV[0]);
  } else {
    for (0..9) {
      printf "%d\t%d\n", &fib($_), &fib($_+10);
    }
  }
}

File: language-testing/fibonacci/fibonacci.lua
#! /usr/bin/env lua

function fib(rotations)
  local old_value = 1
  local new_value = 0

  for i = rotations, 0, -1 do
    new_value, old_value = new_value + old_value, new_value
  end

  return new_value
end

if #arg == 1 then
  print(fib(arg[1]))
else
  for i = 1,10 do
    print(fib(i) .. "\t" .. fib(i+10))
  end
end
